packages/cli/src/commands/console.js

```diff
@@ -9,19 +10,51 @@ export const command = 'console'
 export const aliases = ['c']
 export const description = 'Launch an interactive Redwood shell (experimental)'
 
-const paths = getPaths().api
+const paths = getPaths()
 
-const mapDBToContext = (ctx) => {
-  const { db } = require(path.join(paths.lib, 'db'))
-  ctx.db = db
+const loadPrismaClient = (replContext) => {
+  const { db } = require(path.join(paths.api.lib, 'db'))
+  replContext.db = db
+}
+
+const loadUserConfig = (replContext) => {
+  const userConfigPath = path.join(paths.api.config, 'console')
+  try {
+    const userConfig = require(userConfigPath)
+    Object.assign(replContext, userConfig)
+  } catch (e) {
+    console.error(`Error loading user console config from ${userConfigPath}`)
+    throw e
+  }
+}
+
+const consoleHistoryFile = path.join(paths.generated.base, 'console_history')
+const persistConsoleHistory = (r) => {
+  fs.appendFileSync(
+    consoleHistoryFile,
+    r.lines.filter((line) => line.trim()).join('\n') + '\n',
+    'utf8'
+  )
+}
+
+const loadConsoleHistory = async (r) => {
+  try {
+    const history = await fs.promises.readFile(consoleHistoryFile, 'utf8')
+    history
+      .split('\n')
+      .reverse()
+      .map((line) => r.history.push(line))
+  } catch (e) {
+    // We can ignore this -- it just means the user doesn't have any history yet
+  }
 }
 
 export const handler = () => {
   // Transpile on the fly
   babelRequireHook({
-    extends: path.join(paths.base, '.babelrc.js'),
+    extends: path.join(paths.api.base, '.babelrc.js'),
```

Comment:
Ah, this reminds me that @peterp is planning some re-working of Babel, including removing the default `babelrc.js` in the Redwood project. It will happen within a release or two.

I was going to suggest just removing this line now. However, projects might still have a manual implementation even if we suggest it's not best-practice. In that case, could they manage themselves via the `api/src/lib/console.[ts|js]`?

TBD

Reply:
> Ah, this reminds me that @peterp is planning some re-working of Babel, including removing the default `babelrc.js` in the Redwood project. It will happen within a release or two.

FYI this rework can be found in https://github.com/redwoodjs/redwood/pull/3231